microdnf/include/microdnf/iplugin.hpp

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef MICRODNF_PLUGIN_IPLUGIN_HPP
+#define MICRODNF_PLUGIN_IPLUGIN_HPP
+
+#include "context.hpp"
+
+#include <cstdint>
+
+
+namespace microdnf {
+
+struct Version {
+    std::uint16_t major;
+    std::uint16_t minor;
+    std::uint16_t micro;
+};
+
+static constexpr Version PLUGIN_API_VERSION{.major = 0, .minor = 1, .micro = 0};
+
+class IPlugin {
+public:
+    IPlugin() = default;
+    virtual ~IPlugin() = default;
+
+    IPlugin(const IPlugin &) = delete;
+    IPlugin(IPlugin &&) = delete;
+    IPlugin & operator=(const IPlugin &) = delete;
+    IPlugin & operator=(IPlugin &&) = delete;
+
+    /// Returns the version of the API supported by the plugin. It can be called at any time.
+    virtual Version get_api_version() const noexcept = 0;
+
+    /// Returns the name of the plugin. It can be called at any time.
+    virtual const char * get_name() const noexcept = 0;
+
+    /// Gets the plugin version. It can be called at any time.
+    virtual Version get_version() const noexcept = 0;
+
+    /// Gets the value of the attribute from the plugin. Returns nullptr if the attribute does not exist.
+    /// It can be called at any time.
```

Comment:
The arguments should be documented in the docstring via:
> @param name The name of the attribute to get.

Reply:
OK